### DISCUSSION TITLE
How to get all open editors

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/vscode-discussions#1072] [Extension Development QnA] [answered]
I need to:
- get all open editors
- find the editor that match a certain name
- make the editor visible

Currently, I am stuck at the first step to get all open editors (need to get both visible and invisible ones).

**Comment by r3m0t** [comment, marked as answer]
`TextEditor` object only exists for visible editors. The others are just tabs on `vscode.window.tabGroups`.

**Reply by ghost** [reply]
Thanks for clarification.
What's the API to make a tab visible?
